opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/uploader.rs

```diff
@@ -0,0 +1,212 @@
+        // Using .NET compatible format (matches DateTime.ToString("O"))
+
+        let start_time = format!(
+            "{:04}-{:02}-{:02}T{:02}:{:02}:{:02}.{:07}Z",
+            now.year(),
+            now.month(),
+            now.day(),
+            now.hour(),
+            now.minute(),
+            now.second(),
+            now.nanosecond() / 100 // Convert nanoseconds to 7-digit precision
+        );
+
+        let end_time = format!(
+            "{:04}-{:02}-{:02}T{:02}:{:02}:{:02}.{:07}Z",
+            end_time.year(),
+            end_time.month(),
+            end_time.day(),
+            end_time.hour(),
+            end_time.minute(),
+            end_time.second(),
+            end_time.nanosecond() / 100 // Convert nanoseconds to 7-digit precision
+        );
+
+        // URL encode parameters
+        // TODO - Maintain this as url-encoded in config service to avoid conversion here
+        let encoded_monitoring_endpoint: String =
+            byte_serialize(self.monitoring_endpoint.as_bytes()).collect();
+
+        let encoded_source_identity: String =
+            byte_serialize(self.config.source_identity.as_bytes()).collect();
+
+        // Create a source unique ID - using a UUID to ensure uniqueness
+        let source_unique_id = Uuid::new_v4().to_string();
+
+        // Create the query string
+        format!(
+            "api/v1/ingestion/ingest?endpoint={}&moniker={}&namespace={}&event={}&version={}&sourceUniqueId={}&sourceIdentity={}&startTime={}&endTime={}&format={}&dataSize={}&minLevel={}&schemaIds={}",
+            encoded_monitoring_endpoint,
+            self.moniker,
+            self.config.namespace,
+            event_name,
+            event_version,
+            source_unique_id,
+            encoded_source_identity, // source identity (tenant/role/instance)
+            start_time,
+            end_time,
+            "centralbond/lz4hc", // Format encoding
+            data_size,
+            2, // Min level
+            self.config.schema_ids
+        )
+    }
+
+    /// Uploads data to the ingestion gateway
+    ///
+    /// # Arguments
+    /// * `data` - The encoded data to upload (already in the required format)
+    ///
+    /// # Returns
+    /// * `Result<IngestionResponse>` - The response containing the ticket ID or an error
+    #[allow(dead_code)]
+    pub(crate) async fn upload(
+        &self,
+        data: Vec<u8>,
+        event_name: &str,
+        event_version: &str,
+    ) -> Result<IngestionResponse> {
+        let data_size = data.len();
+        let upload_uri = self.create_upload_uri(data_size, event_name, event_version);
+        let full_url = format!(
+            "{}/{}",
+            self.auth_info.endpoint.trim_end_matches('/'),
+            upload_uri
+        );
+
+        // Send the upload request
+        let response = self
+            .http_client
+            .post(&full_url)
+            .header(header::ACCEPT, "application/json")
```

Comment:
This could be added as a default header while building the Http client?

Reply:
Good point, updated.

Reply:
We can remove this from here now that you have added it as a default header.